### DISCUSSION TITLE
Aktuelle Firmwareversion auslesen

### DISCUSSION THREAD

**Discussion by manny4566** [OpenCCU/OpenCCU#2271] [🇩🇪 Fragen&Antworten] [answered]
Hi,
gibt es einen Befehl mit dem man über die bash die aktuelle Firmwareversion ausgelesen bekommt?
Oder als Alternative die genaue URL um die Help Funktion aufzurufen per curl?
Vielleicht kennt da jemand einen genauen Aufruf. 
Danke

**Comment by jens-maus** [comment, marked as answer]
Was genau willst du damit erreichen?

Um die aktuelle RaspberryMatic in einer SSH Sitzung bzw in der Konsole auszugeben und auch nach einer neueren Version zu checken gibt es das Shell-Kommando…

`checkFirmwareUpdate.sh`

… wenn man aber nur die momentan installierte Version abfragen will reicht mitunter auch einfach

`cat /VERSION`

das gibt dann die installierte Versionsnummer aus.

**Reply by manny4566** [reply]
> Was genau willst du damit erreichen?
> 
> Um die aktuelle RaspberryMatic in einer SSH Sitzung bzw in der Konsole auszugeben und auch nach einer neueren Version zu checken gibt es das Shell-Kommando…
> 
> `checkFirmwareUpdate.sh`
> 
> … wenn man aber nur die momentan installierte Version abfragen will reicht mitunter auch einfach
> 
> `cat /VERSION`
> 
> das gibt dann die installierte Versionsnummer aus.


Ach so einfach. Genau das habe ich gesucht und es übersehen  :D
Danke

Möchte mir ein kleinen Icinga Check schreiben, der mir anzeigt, ob es eine neue Version gibt. Dann muss ich mich nicht anmelden um die Info zu sehen.

Wo wir schon dabei sind. Gibt es auch eine Möglichkeit die Info für Geräte- oder Addon Updates über die Shell abzugreifen?

Oder würde das `checkFirmwareUpdate.sh` auch ausgeben?
Bei `checkAddonUpdates.sh` kommt gerade bei mir kein Output oder ist das normal, wenn es kein Update gibt ?

**Reply by manny4566** [reply]
> Wo wir schon dabei sind. Gibt es auch eine Möglichkeit die Info für Geräte- oder Addon Updates über die Shell abzugreifen?
> 
> Oder würde das `checkFirmwareUpdate.sh` auch ausgeben? Bei `checkAddonUpdates.sh` kommt gerade bei mir kein Output oder ist das normal, wenn es kein Update gibt ?

@jens-maus 
Hi, könntest du mir die Fragen noch beantworten? Dann können wir ein Haken dran machen :) Danke

**Reply by jens-maus** [reply]
Schau dir doch beide Shell-Skripte einfach nach und bau das nach was diese tun. Im Grunde machen die nur HTTP Requests an gewisse Server, usw. um nach neuen Versionen Ausschau zu halten. Empfinde ich aber trotzdem als overkill hier irgendwelche Incinga/CheckMK Checks bauen zu wollen um nach Firmware oder Addon-Updates ausschau halten zu wollen. Damit belastest du nur unnötig die Webserver der Anbieter und sorgst ggf. dafür das der Dienst dann ggf. einfach mal eingestellt wird weil nur diese UpdateCheck abfragen dauernd einprasseln von tausenden von Nutzern.

**Reply by Weseler485** [reply]
> Was genau willst du damit erreichen?
> 
> Um die aktuelle RaspberryMatic in einer SSH Sitzung bzw in der Konsole auszugeben und auch nach einer neueren Version zu checken gibt es das Shell-Kommando…
> 
> `checkFirmwareUpdate.sh`

Ich bekomme ja auf der Startseite angezeigt, ob es eine neue Version gibt oder nicht. Wird diese Abfrage bei jedem Login neu gemacht oder wird das Ergebnis irgendwo zwischengespeichert?

**Reply by jens-maus** [reply]
> Ich bekomme ja auf der Startseite angezeigt, ob es eine neue Version gibt oder nicht. Wird diese Abfrage bei jedem Login neu gemacht oder wird das Ergebnis irgendwo zwischengespeichert?

Diese Abfrage auf der Startseite bzgl. verfügbares RaspberryMatic Update findet komplett via Javascript im Browser-Kontext statt. Wird also jedesmal frisch gemacht wenn du deinen Browser auf deine RaspberryMatic zugreifen lässt. Ergo funktioniert diese auch selbst wenn das RaspberryMatic selbst keinen Internetzugriff für diese Prüfung hat. `checkFirmwareUpdate.sh` ist davon unabhängig.

**Comment by oradke** [comment]
Warum abonnierst Du nicht einfach die Updates via GitHub? Jens schick immer eine ausführliche Mail wenn es ein neues Release gibt.

Von: manny4566 ***@***.***>
Gesendet: Dienstag, 28. März 2023 16:55
An: jens-maus/RaspberryMatic ***@***.***>
Cc: Subscribed ***@***.***>
Betreff: Re: [jens-maus/RaspberryMatic] Aktuelle Firmwareversion auslesen (Discussion #2271)


Was genau willst du damit erreichen?

Um die aktuelle RaspberryMatic in einer SSH Sitzung bzw in der Konsole auszugeben und auch nach einer neueren Version zu checken gibt es das Shell-Kommando...

checkFirmwareUpdate.sh

... wenn man aber nur die momentan installierte Version abfragen will reicht mitunter auch einfach

cat /VERSION

das gibt dann die installierte Versionsnummer aus.

Ach so einfach. Genau das habe ich gesucht und es übersehen :D
Danke

Möchte mir ein kleinen Icinga Check schreiben, der mir anzeigt, ob es eine neue Version gibt. Dann muss ich mich nicht anmelden um die Info zu sehen.

Wo wir schon dabei sind. Gibt es auch eine Möglichkeit die Info für Geräte- oder Addon Updates über die Shell abzugreifen?

Oder würde das checkFirmwareUpdate.sh auch ausgeben?
Bei checkAddonUpdates.sh kommt gerade bei mir kein Output oder ist das normal, wenn es kein Update gibt ?

-

**Reply by manny4566** [reply]
> Warum abonnierst Du nicht einfach die Updates via GitHub? Jens schick immer eine ausführliche Mail wenn es ein neues Release gibt. Von: manny4566 ***@***.***> Gesendet: Dienstag, 28. März 2023 16:55 An: jens-maus/RaspberryMatic ***@***.***> Cc: Subscribed ***@***.***> Betreff: Re: [jens-maus/RaspberryMatic] Aktuelle Firmwareversion auslesen (Discussion #2271) Was genau willst du damit erreichen? Um die aktuelle RaspberryMatic in einer SSH Sitzung bzw in der Konsole auszugeben und auch nach einer neueren Version zu checken gibt es das Shell-Kommando... checkFirmwareUpdate.sh ... wenn man aber nur die momentan installierte Version abfragen will reicht mitunter auch einfach cat /VERSION das gibt dann die installierte Versionsnummer aus. Ach so einfach. Genau das habe ich gesucht und es übersehen :D Danke Möchte mir ein kleinen Icinga Check schreiben, der mir anzeigt, ob es eine neue Version gibt. Dann muss ich mich nicht anmelden um die Info zu sehen. Wo wir schon dabei sind. Gibt es auch eine Möglichkeit die Info für Geräte- oder Addon Updates über die Shell abzugreifen? Oder würde das checkFirmwareUpdate.sh auch ausgeben? Bei checkAddonUpdates.sh kommt gerade bei mir kein Output oder ist das normal, wenn es kein Update gibt ? - Reply to this email directly, view it on GitHub<[#2271 (reply in thread)](https://github.com/jens-maus/RaspberryMatic/discussions/2271#discussioncomment-5455295)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHCHQNYRF4PPUJA7RO42WPLW6L3VJANCNFSM6AAAAAAWIFOH5A>. You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

Hab halt ein lokales Monitoring über Icinga2 und möchte dort direkt die Info sehen, wenn es ein Update vorhanden ist. 
Da bringt mir eine Mail nichts.
